### DISCUSSION TITLE
mpremote copy doesn't work

### DISCUSSION THREAD

**Discussion by Vellug** [micropython/micropython#16423] [Zephyr]
Hello everyone,

I'm a student currently working on my bachelorthesis. So I'm not too experienced an hopefully my questions aren't too stupid. 

(1) My first question is: Is there something like a list where you can see which defines in mpconfigport you have to set when using per example mpremote?

(2) I'm trying to use the mpremote cp command to load a file from my laptop to my device but I get this error message:
```py
$ mpremote cp set_led.py :
cp set_led.py :
mpremote: Error with transport:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: 2
```

Note: I can use the file with "mpremote run" without an error and can create a file on the device filesystem with "mpremote mkdir"

Due to the "stdin" in the error message I thought I need to set in the mpconfigport.h the MICROPY_PY_SYS_STDFILES to true. But if I do this I get an error during the build. Someone already had the same issue "#5162" and I tried to apply the fix from dpgeorge with commit 2863dcd but I get the following message in the function "mp_hal_std_poll" (mphalport.c):

```c
struct "_mp_state_vm_t" has no field "board_stdio_uart"
```

Is this something I have to define in the struct? Doesn't seem to me like this struct is something that I should change as a user.

**Comment by peterhinch** [comment]
`mpremote cp` should work without any special actions. I have seen this error on specific combinations of PC and target hardware - I suspect a USB issue. Options are to try another PC or try [rshell](https://github.com/dhylands/rshell).

**Reply by Vellug** [reply]
Thank you for your reply. 

rshell didn't work aswell. But I found out that the problem is somewhere in my own micropython port depending on some changes that I have made, because with the "fresh" zephyr port it works.
